### PR TITLE
Simplify logic #167 - Changes to view.jsx

### PR DESCRIPTION
**Pull request by sautumn**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Modularized the SideBar component by using DRY practices. Made the function `getNavLink` to render each navLink that is now part of a navLinks array. Also removed unneeded import and props. 

Does this close any currently open issues?
------------------------------------------
Links issue #167 

Where has this been tested?
---------------------------
**Operating System:** MacOS, Electron

### CODE REVIEW

File: src/renderer/components/sidebar/view.jsx [hunk 1]
```diff
@@ -1,56 +1,68 @@
...
 import Icon from '@mdi/react'
 import classnames from 'classnames'
-import navigate from '@/utils/navigate'
-import Button from '@/components/common/button'
-import * as icons from '@/constants/icons'
+import {
+  HOME as home,
+  PLAYLIST_PLUS as playlistPlus,
+  HEART_OUTLINE as heart,
+} from '@/constants/icons'
```
Comment (btzr-io) [inline src/renderer/components/sidebar/view.jsx]: Nice!

File: src/renderer/components/sidebar/view.jsx [hunk 2]
```diff
@@ -1,56 +1,68 @@
...
+)
 
 class SideBar extends React.PureComponent {
   constructor(props) {
     super(props)
   }
 
-  render() {
-    const { doNavigate, navigation, favorites } = this.props
+  getNavLink = ({ path, label, icon, idx }) => {
```
Comment (btzr-io) [inline src/renderer/components/sidebar/view.jsx]: Is this function really necessary?

Comment (btzr-io) [reply src/renderer/components/sidebar/view.jsx]: Did you test this? Other than that looks good :+1:

Comment (sautumn) [reply src/renderer/components/sidebar/view.jsx]: I tested it locally. 

The getNavLinks reduces typing <NavLink> over and over which is better practice. https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

**Pull request merged**